### ISSUE TITLE
Support postponed evaluation of annotations for classes that contain attributes with the class as its type

### ISSUE THREAD

**Issue by Ninja3047** [PyCQA/pylint#2847]
### Steps to reproduce
```
# pylint: disable=missing-docstring
# pylint: disable=too-few-public-methods

from __future__ import annotations

class ExampleSelf:
    next: ExampleSelf

```

### Current behavior
```
************* Module self_reference
self_reference.py:7:10: E0602: Undefined variable 'ExampleSelf' (undefined-variable)

--------------------------------------------------------------------
Your code has been rated at -6.67/10 (previous run: -6.67/10, +0.00)
```
### Expected behavior
No error
mypy returns no error and python doesn't throw a name error

### pylint --version output
```
pylint 2.4.0-dev0
astroid 2.2.5
Python 3.7.2 (default, Mar  5 2019, 12:58:23) 
[GCC 8.2.0]
```
### Related issues
#2069 #2838

**Comment by PCManticore** [comment]
Thanks for the report @Ninja3047 !
